### PR TITLE
Calculate the difference between the subject and the expected nearby time

### LINKED ISSUE #2012: Update error message at `DateTimeAssertions.BeCloseTo()` to include the calculated time range
**Before you file a bug, have you:**

* Tried upgrading to newest version of Fluent Assertions, to see if your issue has already been resolved and released?
* Checked existing open *and* closed [issues](https://github.com/fluentassertions/fluentassertions/issues?utf8=%E2%9C%93&q=is%3Aissue), to see if the issue has already been reported?
* Tried reproducing your problem in a new isolated project?
* Read the [documentation](https://fluentassertions.com/introduction)?
* Considered if this is a general question and not a bug?. For general questions please use [StackOverflow](https://stackoverflow.com/questions/tagged/fluent-assertions?mixed=1).

### Description

Now the error message looks like this:
`Expected logs.First().ChangedAt to be within 15s from <2022-10-13 06:44:06.8277343>, but found <2022-10-13 06:43:51>.`

But with this line I have to calculate the difference between the two `DateTime`s in case I want to know the exact difference.

### Complete minimal example reproducing the issue

This is not an issue inside FA, but a proposal to state more precisely what the time differnece was.

### Expected behavior:

Something like:
`Expected logs.First().ChangedAt to be within 15s from <2022-10-13 06:44:06.8277343>, but found <2022-10-13 06:43:51>, which is XXXs.`

### Actual behavior:

The calculated difference is not displayed.

### Versions

* Which version of Fluent Assertions are you using? 6.7.0
* Which .NET runtime and version are you targeting? .net6.0 and .net7.0

### Additional Information

-

### PR DESCRIPTION
**Pull request by IT-VBFK**
Closes #2012 

## IMPORTANT 

* [x] The code complies with the [Coding Guidelines for C#](https://www.csharpcodingguidelines.com/).
* [x] The changes are covered by unit tests which follow the Arrange-Act-Assert syntax and the naming conventions such as is used [in these tests](../tree/develop/Tests/FluentAssertions.Equivalency.Specs/MemberMatchingSpecs.cs#L51-L430).
* [ ] If the PR adds a feature or fixes a bug, please update [the release notes](../tree/develop/docs/_pages/releases.md) with a functional description that explains what the change means to consumers of this library, which are published on the [website](https://fluentassertions.com/releases).
* [ ] If the PR changes the public API the changes needs to be included by running [AcceptApiChanges.ps1](../tree/develop/AcceptApiChanges.ps1) or [AcceptApiChanges.sh](../tree/develop/AcceptApiChanges.sh).
* [ ] If the PR affects [the documentation](../tree/develop/docs/_pages), please include your changes in this pull request so the documentation will appear on the [website](https://www.fluentassertions.com/introduction).

### CODE REVIEW

File: Src/FluentAssertions/Primitives/DateTimeAssertions.cs [hunk 1]
```diff
@@ -168,12 +168,18 @@ public AndConstraint<TAssertions> Be(DateTime? expected, string because = "", pa
         long distanceToMaxInTicks = (DateTime.MaxValue - nearbyTime).Ticks;
         DateTime maximumValue = nearbyTime.AddTicks(Math.Min(precision.Ticks, distanceToMaxInTicks));
 
+        var difference = (Subject - nearbyTime)?.Duration();
```
Comment (IT-VBFK) [inline Src/FluentAssertions/Primitives/DateTimeAssertions.cs:171]: `...Duration()`: 
I was not sure what fits better: the "real" difference (meaning this also could be negative) or the absolute values..

Comment (dennisdoomen) [reply Src/FluentAssertions/Primitives/DateTimeAssertions.cs:171]: In this case, the value can be off in both directions, so an absolute value is enough.

File: Src/FluentAssertions/Primitives/DateTimeOffsetAssertions.cs [hunk 2]
```diff
@@ -304,12 +304,18 @@ public DateTimeOffsetAssertions(DateTimeOffset? value)
         long distanceToMaxInTicks = (DateTimeOffset.MaxValue - nearbyTime).Ticks;
         DateTimeOffset maximumValue = nearbyTime.AddTicks(Math.Min(precision.Ticks, distanceToMaxInTicks));
 
+        var differnece = (Subject - nearbyTime)?.Duration();
```
Comment (dennisdoomen) [inline Src/FluentAssertions/Primitives/DateTimeOffsetAssertions.cs:307]: 🔧 Please use the explicit type here.

**Review by dennisdoomen — CHANGES_REQUESTED** [review]
Can you also update the `releases.md` file?

File: Src/FluentAssertions/Primitives/DateTimeAssertions.cs (hunk 1 above)
Comment (IT-VBFK) [reply Src/FluentAssertions/Primitives/DateTimeAssertions.cs:171]: 👍🏻

**Comment by IT-VBFK** [comment]
> Can you also update the `releases.md` file?

Ok
Wasn't sure wheter this change is enough to be mentioned in the official release notes :)

File: Src/FluentAssertions/Primitives/DateTimeOffsetAssertions.cs (hunk 2 above)
Comment (jnyrup) [inline Src/FluentAssertions/Primitives/DateTimeOffsetAssertions.cs]: differnece -> difference.

**Review by dennisdoomen — APPROVED** [review]

File: Src/FluentAssertions/Primitives/DateTimeAssertions.cs [hunk 3]
```diff
@@ -168,12 +168,18 @@ public AndConstraint<TAssertions> Be(DateTime? expected, string because = "", pa
...
             .BecauseOf(because, becauseArgs)
-            .FailWith("Expected {context:the date and time} to be within {0} from {1}{reason}, but found {2}.",
-                precision,
-                nearbyTime, Subject);
+            .WithExpectation("Expected {context:the date and time} to be within {0} from {1}{reason}", precision, nearbyTime)
+            .ForCondition(Subject is not null)
+            .FailWith(", but found <null>.")
+            .Then
+            .ForCondition((Subject >= minimumValue) && (Subject <= maximumValue))
+            .FailWith(", but {0} is off by {1}.", Subject, difference)
```
Comment (jnyrup) [inline Src/FluentAssertions/Primitives/DateTimeAssertions.cs:180]: Should this be in past tense, i.e. "was off by"? 🤔

Comment (IT-VBFK) [reply Src/FluentAssertions/Primitives/DateTimeAssertions.cs]: Hmm.. 

fair point :)

**Review by jnyrup — APPROVED** [review]

**Pull request merged**